ci_action/main.py

```diff
@@ -103,18 +103,32 @@ def get_environment_config():
     self_test = os.environ.get('CI_SELF_TEST', 'false').lower() == 'true'
     test_script = os.environ.get('TEST_SCRIPT', 'run_tests.sh')
 
-    # Test dependencies as bundle items
-    test_deps = [d.strip() for d in os.environ.get('UNITTEST_BUNDLE_DEPENDENCIES', '').split(' ')]
-    filtered_test_deps = []
-    for td in test_deps:
-        if td:
-            filtered_test_deps.append(td)
-
     # Get the target project name. If not passed explicitly, use the repo name.
     target_project_name = os.environ.get('TARGET_PROJECT_NAME', '')
     if not target_project_name.strip():
         target_project_name = repo_name
 
+    # Collect the test dependencies from the environment variables.
+    unittest_deps_env = os.environ.get('UNITTEST_DEPENDENCIES', '').strip()
+    unittest_deps = [d.strip() for d in unittest_deps_env.split(' ') if d.strip()]
+    test_deps_env = os.environ.get('TEST_DEPENDENCIES', '').strip()
+    test_deps = [d.strip() for d in test_deps_env.split(' ') if d.strip()]
+
+    test_strategy = os.environ.get('TEST_STRATEGY', 'all').lower()
+    if test_strategy not in ['all', 'integration', 'unit']:
+        raise ValueError('Option "test_strategy" must be one of "ALL", "INTEGRATION", or "UNIT".')
+
+    # Handle deprecated config option 'unittest_dependencies'.
+    if test_deps and unittest_deps:
+        raise ValueError('Config option "unittest_dependencies" and '
+                         '"test_dependencies" cannot be set simultaneously.')
+    elif unittest_deps:
+        test_deps = unittest_deps
+        test_strategy = 'all'
+
+    # Ensure that the test dependencies include the target project.
+    test_deps = [d for d in set(test_deps + [target_project_name])]
```

Comment:
Note, because the "unittest_deps" and "test_deps" are mutually exclusive (with unittest being supported for a deprecation period before I delete it), we only need to pass forward the "true" value of this list based on the selection logic above. In a month I'll update this to surface an error when `UNITTEST_DEPENDENCIES` is used, and in a few more months I'll delete the old option entirely.